spec/models/user_spec.rb

```diff
@@ -0,0 +1,94 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  let!(:user) {FactoryBot.build :user}
+  let!(:user_fail) {FactoryBot.build :user, name: nil}
```

Comment:
khác nhau giữa ! và k ! là gì e ei

Reply:
- "let" chỉ có giá trị trong mỗi block example "it", nó sẽ lại được khởi tạo lại đúng 1 lần duy nhất trong block đó (lazy evaluation).
- Còn "let!" bản chất là được thực hiện trong 1 khối "before" ngầm, tức là giá trị sẽ được tính toán và cache lại trước khi khối "it" được thực thi.